### DISCUSSION TITLE
How to edit custom status?

### DISCUSSION THREAD

**Discussion by DeadLyBro** [rigwild/discord-self-bot-console#36] [Q&A] [answered]
If can we edit our custom status with API, can you share how to do and if its possible please update repo ;) @rigwild 

![Screenshot_3](https://user-images.githubusercontent.com/31214077/188722785-3ec9641c-cdfe-4067-b318-0e1f2ce31e0b.png)

I write there "Hey!".

**Comment by rigwild** [comment]
This is not documented but I figured out you can edit the `bio` user profile key. The client does not get the update, and I'm not sure with other users. The change is visible on client restart though 🤔

Updated the script to allow to edit any profile property.

```js
// api.editCurrentUser(username?, bio?, body = {})
await api.editCurrentUser(null, 'hellooooo')
```

➡️ [Listen to events, do some advanced stuff
](https://github.com/rigwild/discord-self-bot-console#listen-to-events-do-some-advanced-stuff)

**Reply by DeadLyBro** [reply]
No, i don't mean bio, i mean custom status.

**Comment by DeadLyBro** [comment, marked as answer]
Edit: rigwild updated `index.js` try new codes.


I write it myself; 

Add any line on API; 
```js
customStatus: (emoji_id, emoji_name, expires_at, text) => apiCall(`/users/@me/settings`, { custom_status: { 'emoji_id': emoji_id, 'emoji_name': emoji_name, 'expires_at': expires_at, 'text': text } }, 'PATCH'),
```

This is how to use; 

```js
{
	await api.customStatus("emoji_id", "👍", "ISO8601 Expire time", "Your text") 
	// If you're not have Nitro change "emoji_id" to null. 
	// ISO8601 Expire Time Example: 2025-03-01T13:00:00Z
	// Custom Status for 30 Second :     new Date(Date.now() + 30 * 1000).toJSON() 
	// Thanks to Rigwild.
}
``` 

This is example; 
```js
{
	await api.customStatus(null, "👍", new Date(Date.now() + 30 * 1000).toJSON(), "My Custom Text")
}
```

Attention: If you're making this auto, please set delay to 5 or 10 second if you don't want to be ban for unusual activity.


I hope this works, please try this @rigwild .

**Reply by rigwild** [reply]
It works! Added to the script ✌️ Thanks!!

```js
api.setCustomStatus(emojiId?, emojiName?, expiresAt?, text?)
api.deleteCustomStatus()
```

Set a custom status for 30s:

```js
await api.setCustomStatus(null, "👍", new Date(Date.now() + 30 * 1000).toJSON(), "hello")
```

**Reply by DeadLyBro** [reply]
You're welcome dude ♥
